src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4401,6 +4402,39 @@ private BoundExpression BindClassCreationExpression(ObjectCreationExpressionSynt
             }
         }
 
+#nullable enable
+        /// <summary>
+        /// Helper method to create a synthesized constructor invocation.
+        /// </summary>
+        private BoundExpression MakeClassCreationExpression(
```

Comment:
>MakeClassCreationExpression [](http://example.com/codeflow?start=32&length=27)

The naming might be somewhat confusing. The helper isn't just for classes right? Structures are good too? You are probably follwong the pattern of BindClassCreationExpression name, however that function wasn't meant to be used as a general helper. Consider using name that simply indicates that an instance is created by using constructor. Then arguments start making more sense, etc. "MakeConstructorInvocation"?